src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -804,10 +865,20 @@ static ColumnWithTypeAndName readNonNullableColumnFromArrowColumn(
                     /// ORC doesn't support Decimal256 as separate type. We read and write it as binary data.
                     case TypeIndex::Decimal256:
                         return readColumnWithBigNumberFromBinaryData<ColumnDecimal<Decimal256>>(arrow_column, column_name, type_hint);
+                    case TypeIndex::Object:
+                        if (settings.enable_json_parsing)
+                            return readColumnWithJSONData<arrow::BinaryArray>(arrow_column, column_name);
+                        [[fallthrough]];
                     default:
                         break;
                 }
             }
+
+            if (settings.enable_json_parsing and isColumnJSON(arrow_field))
```

Comment:
```suggestion
            if (settings.enable_json_parsing && isColumnJSON(arrow_field))
```

Please, don't use `and`/`or`, use standart `&&`/`||`